### PR TITLE
Fixed a bug that was causing the UX to reset visual editor trigger conditions to their default values when a trigger name contained periods.

### LINKED ISSUE #203: [BUG] Period (.) at the beginning or end of a trigger name resets trigger conditions when editing monitor
**Describe the bug**
Having a `.` at the beginning or end of a trigger name causes the trigger condition values to reset when editing the monitor in the UX. 

**To Reproduce**
1. Go to `Create monitor`
2. Define a `Per query monitor`
3. Create a trigger with a name that includes a `.` as the first or last character; e.g., `.trigger` or `trigger.`
4. Create the monitor
5. Go to the edit monitor page
6. The trigger condition will reset to the default values of `IS ABOVE 10000`

**Expected behavior**
The UX should display the configured trigger condition; not the defaults.

**Plugins**
`Alerting` and `alertingDashboards`.

**Screenshots**

**Host/Environment (please complete the following information):**
 - OS: IOS
 - Version 10.15.7 

**Additional context**
Related issues from the ODFE alerting plugin repo:
https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/issues/152
https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/issues/199
https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/issues/207

### PR DESCRIPTION
**Pull request by AWSHurneyt**
Signed-off-by: AWSHurneyt <hurneyt@amazon.com>

### Description
Fixed a bug that was causing the UX to reset visual editor trigger conditions to their default values when a trigger name contained periods.

```js
fields = {
  Cusip: {
    ".trigger": 123,
    "trigger": 56
  }
};

_.get(fields, "Cusip..trigger", 11) 
# returns 56
# seems lodash will squash the .. to .
```
 
### Issues Resolved
#203 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by bowenlan-amzn** [comment]
@brijos would like have your opinion on whether to support a trigger name with a dot at beginning or the end. like `.<trigger_name>.` Thanks.

File: public/pages/CreateTrigger/containers/CreateTrigger/utils/triggerToFormik.js [hunk 1]
```diff
@@ -118,37 +124,46 @@ export function bucketLevelTriggerToFormik(trigger, monitor) {
...
     FORMIK_INITIAL_TRIGGER_VALUES.thresholdValue
   );
+
+  const adTriggerMetadata = _.get(triggersUiMetadata[name], 'adTriggerMetadata', {});
+  /*
+    If trigger type doesn't exist fallback to query trigger with following reasons
+    1. User has changed monitory type from normal monitor to AD monitor.
+    2. User has created / updated from API and visiting OpenSearch Dashboards to do other operations.
+   */
+  const triggerType = _.get(adTriggerMetadata, 'triggerType', TRIGGER_TYPE.ALERT_TRIGGER);
```
Comment (bowenlan-amzn) [inline public/pages/CreateTrigger/containers/CreateTrigger/utils/triggerToFormik.js:145]: This part seems to be same as above, do we need it?

Comment (AWSHurneyt) [reply public/pages/CreateTrigger/containers/CreateTrigger/utils/triggerToFormik.js:145]: Sorry, which lines are you referring to?

**Review by bowenlan-amzn — APPROVED** [review]

File: public/pages/CreateTrigger/containers/CreateTrigger/utils/triggerToFormik.js (hunk 1 above)
Comment (bowenlan-amzn) [reply public/pages/CreateTrigger/containers/CreateTrigger/utils/triggerToFormik.js:145]: Realised they are in 2 functions, so that's fine.

**Review by lezzago — APPROVED** [review]

**Pull request merged**